operator-common/src/main/java/io/strimzi/operator/common/Util.java

```diff
@@ -158,26 +185,50 @@ public static File createFileStore(String prefix, String suffix, byte[] bytes) {
      * Decode binary item from Kubernetes Secret from base64 into byte array
      *
      * @param secret    Kubernetes Secret
-     * @param key       Key which should be retrieved and decoded
+     * @param field     Field which should be retrieved and decoded
      * @return          Decoded bytes
      */
-    public static byte[] decodeFromSecret(Secret secret, String key) {
-        return Base64.getDecoder().decode(secret.getData().get(key));
+    public static byte[] decodeBase64FieldFromSecret(Secret secret, String field) {
+        return Optional.ofNullable(secret)
+                .map(Secret::getData)
+                .map(data -> data.get(field))
+                .map(value -> Base64.getDecoder().decode(value))
+                .orElseThrow(() -> {
+                    String name = Optional.ofNullable(secret)
+                            .map(Secret::getMetadata)
+                            .map(ObjectMeta::getName)
+                            .orElse("unknown");
+                    String namespace = Optional.ofNullable(secret)
+                            .map(Secret::getMetadata)
+                            .map(ObjectMeta::getNamespace)
+                            .orElse("unknown");
+                    return Util.missingDataInSecretException(namespace, name, field);
+                });
```

Comment:
Is this really an efficient and readable way to handle this? I would find using `if` constructs much easier to read and understand what exactly does this actually do.

Reply:
It's probably a bit over the top, given that I'm assuming if the secret is not null then we can assume the name, namespace and data should also not be null, so can probably change to something like:

```
        if (secret == null) {
            throw new RuntimeException("Cannot fetch data from null secret");
        }
        String data = secret.getData().get(field);
        if (data != null) {
            return Base64.getDecoder().decode(data);
        } else {
            throw Util.missingDataInSecretException(secret.getMetadata().getNamespace(), secret.getMetadata().getName(), field);
        }
```

Reply:
I've updated this to be simpler and make use of Objects.requireNonNull